README.md
# The Coding Quiz Readme file
By Austin Keener <br>
<i>These are the terms of installation for this code.<i>
## Table of Contents
### 1. Usage <br>
### 2. License<br>

link: https://austingraphicandweb.github.io/the-coding-quiz/
<hr>

## Screenshot

![screenshot](/assets/Screen-Shot.png)

<hr>


## Usage
<i>This project is geared towards using javascript as well as my own html and css to create a quiz that users can interact with. The game is played out over 60 seconds. This is how much time you have to get the answers correct. You loose 15 seconds every time an answer is marked wrong.<i>
<hr>

## Licensing
<i>
MIT License

Copyright (c) [2020] [Austin Graphic and Web]

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.<i>
<hr>

changes.md
These are a list of the edits that I am making.

## Readme file populated
## Added boiler plate html
## Linked js and css to html document
## using semantic tags I created the header, main, and section elements. 
## I put a spot for viewing highscores and timer in the header. 
## I used various child elements to create the title, instructions, quiz sections for js, button to start quiz, and section to track the score through js
## Used DOM to target quiz, start-btn, and score-tracker by id
## Updated spelling on index.html 
## Added an event listener to js
## Additional variable to store questions and answers in a object method
## Creation of a function to run the quiz, and a function to show the score of the quiz
## Button calling the quiz
## Added css properties to look similar to example gif

assets/script/javascript.js
/**
 * The Coding Game Javascript File
 */

// Global Elements are used throughout the js script. I am getting elements from my html page.
// var score = document.getElementById("score-tracker");
// var homeScreen = document.getElementById("home-screen");
// var submit = document.getElementById("submit");
// var goBack = document.getElementById("go-back");
// var clear = document.getElementById("clear-score");
var startBtn = document.getElementById("start-btn");
var questionElement = document.getElementById("questions");
var theTimer = document.getElementById("timer");
var enterInitials = document.getElementById("score");
var highScores = document.getElementById("highscores");
var currentQuestion = 0;
var timeLeft = 60;

// Start game. A function that clears the homescreen and calls the quiz and timer.
function startGame(){

    // I am creating a function that starts the game when button is clicked. First, I call the document and get the "welcome" div element by id. Then I set the id to .innerHTML so that I can replace the content. I replace the content with "" so that there is nothing inside of the div.
    document.getElementById("home-screen").innerHTML = "";
    theQuiz();
    timer();
}

// Timer. The timer was based off of an activity done in class. I am using a function that contains a variable describing the time left and another variable that defines the interval of time that is being counted down. the last part of this function is a conditional statement that clears the interval after it has reached zero.
function timer(){
    var interval = setInterval(function()
    {theTimer.textContent = timeLeft + " seconds left.";
    timeLeft--;

    if (timeLeft <= 0 || qna >= [4]){
        theTimer.textContent = "";
        clearInterval(interval);
        // qna[currentQuestion].addEventListener("click", function() {
        //     // theQuiz.textContent="";
        //     alert("making sure that this is working");
        // })
    }
    
    }, 1000);
    
}
// Question and answers
var qna = [
    {
        question: "Commonly used data types DO NOT include:",
        choices:["strings", "booleans", "alerts","numbers"],
        answer: "alerts"
    },

    {
        question: "The condition in an if / else statement is enclosed within ______.",
        choices: ["quotes", "curly brackets", "parentheses", "square brackets"],
        answer: "curly brackets"
    },

    {
        question: "Arrays in JavaScript can be used to store ______.",
        choices: ["numbers and strings", "other arrays", "booleans", "all of the above"],
        answer: "all of the above"
    },

    {
        question: "String values must be enclosed within ______ when being assigned to variables.",
        choices: ["commas", "curly brackets", "quotes", "parentheses"],
        answer: "quotes"
    },

    {
       question:"A very useful tool used during development and debugging for printing content to the debugger is:",
       choices: ["JavaScript", "terminal/bash", "for loops", " console.log"],
       answer: "console.log"
    }
];

// The quiz
function theQuiz(){
    var h2El = document.createElement("h2");
    questionElement.textContent = "";
    questionElement.style.display = "block";
    h2El.textContent = qna[currentQuestion].question;
    questionElement.appendChild(h2El);
    // this for loop is to cycle through the questions and answers.
    for (i = 0; i < qna[currentQuestion].choices.length; i++) {
        var buttonEl = document.createElement("button");
        questionElement.append(buttonEl);
        buttonEl.textContent = qna[currentQuestion].choices[i];
        buttonEl.style.display= "block";
        buttonEl.style.margin= "10px";
        buttonEl.addEventListener("click", quizButtons);
        }
    if (qna[currentQuestion] === [4]) {
        qna[currentQuestion].answer.onClick = questionElement.textContent = "";
    }
   
}


function quizButtons(){
    if (qna[currentQuestion].answer === event.target.textContent){
            alert("correct!");

    } else {
        timeLeft-=15;
    };
    currentQuestion++;
    theQuiz();
    // if(currentQuestion === qna.length){
    //     theQuiz.textContent = "";
    //     theScore();
    // }
}

// Function to get the results of the quiz. After the last answer is chosen, the quiz is cleared, "game over is displayed", my final score is displayed, and a box for entering in my initials is displayed.
function theScore(){
    // the.textContent = "";
    var h1El = document.createElement("h1");
    var h3El = document.createElement("h3");
    var h3El = document.createElement("h3");
    var formBox = document.createElement("form");
    formBox.textContent = "";
    h1El.textContent = "Game Over";
    h3El.textContent = "Please enter your score";
    enterInitials.appendChild(h1El);
    enterInitials.appendChild(h3El);
}

// setting up local storage that will run along with the quiz to store the results....
// in this first line of code I am declaring a variable that is equal to my object variable containing my questions
var storage = JSON.stringify(qna);
// now I am going to create a variables that takes apart the stringify and makes it look like a regular object again
var storageOrganized = JSON.parse(localStorage.getItem("qna"));
// now I am going to set up the local storage so that it displays the value of 'qna' in the console when the web page is opened
localStorage.setItem("qna", storage);

// After my initials have been entered in and the "submit" button has been clicked, the highscore page appears and lets me see my score and the score of anyone else that has taken the quiz also. The information is stored in local storage.
function highScores(){
    var highscoreHead = document.createElement("h1");
    highScores.textContent = "";
    highscoreHead.textContent = "Highscores";

}

startBtn.addEventListener("click", startGame);

 